### ISSUE TITLE
Possibility to add JWT authentication ?

### ISSUE THREAD

**Issue by cyril-colin** [calvinckho/capacitor-jitsi-meet#26]
Hi and thanks you for this useful plugin !

I am thinking about using JWT authentication on my self-hosted instance of jitsi thanks to this guide https://github.com/jitsi/lib-jitsi-meet/blob/master/doc/tokens.md

So, I need to set a token on client side. It is easy with the lib-jitsi-meet (for web implementation only): I just have to add it to the BOSH url as query param.
But with this capacitor plugin, I can't see any options to add the token.

I can see in Jitsi SDK that there is a property for that, but this property is never set in your code
- SDK iOS : https://github.com/jitsi/jitsi-meet/blob/master/ios/sdk/src/JitsiMeetConferenceOptions.m
- SDK Android : https://github.com/jitsi/jitsi-meet/blob/master/android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetConferenceOptions.java

My question is : Do you think I can just add the JWT as query param in your serverUrl exposed option ? If not, do you think you can add this as a new feature ?

Thanks !

**Comment by calvinckho** [comment]
what is the name of the variable for the JWT token in the native codes?

**Comment by cyril-colin** [comment]
Thank you for your response.

The name of the variable is "token" in android and iOS codes :

Java :  https://github.com/jitsi/jitsi-meet/blob/master/android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetConferenceOptions.java
```java
/**
 * JWT token used for authentication.
 */
private String token;
```

iOS : https://github.com/jitsi/jitsi-meet/blob/master/ios/sdk/src/JitsiMeetConferenceOptions.m
```objective-c
// Init code
_token = nil;

// other code
if (_token != nil) {
    urlProps[@"jwt"] = _token;
}
```

**Comment by cyril-colin** [comment]
Here is an example of what it could looks like : https://github.com/calvinckho/capacitor-jitsi-meet/pull/27

It is not yet completly tested because I have some troubles with my jitsi server.

I keep you informed.

**Comment by DanielnetoDotCom** [comment]
This is very important, any way to resolve this?

I notice on the Cordova ones they submit the JWT with the domain URL `domain.url?jwt=...`

but here looks like something clean the domain and remove the question mark `?` and the equals sign `=`.

**Comment by calvinckho** [comment]
@DanielnetoDotCom and @cyril-colin, I am busy with another project and due date at hand. The earliest I can get to this will be early next week.

**Comment by DanielnetoDotCom** [comment]
Thanks, I appreciate thanks.

**Comment by DanielnetoDotCom** [comment]
@calvinckho are you available for a paid job?

If yes please email me at danielneto.com@gmail.com

**Comment by cyril-colin** [comment]
@calvinckho  Hi :) Have you more time to think about my PR ? I have tested it on android and iOS and it's working !

**Comment by calvinckho** [comment]
@cyril-colin yes, I released 1.5.0 using your code. Can you install and it with token and see if you are able to authenticate with your video bridge?
